### PR TITLE
Validate machineconfig uniqueness when ref name changes during upgrade

### PR DESCRIPTION
**Pull request by jiayiwang7**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/567#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (g-gaston) [inline pkg/providers/vsphere/vsphere.go:931]: nit: personal opinion, this kind of acronyms make the code way more difficult to read. I believe using thing like `capi` is ok because it's a shared term in the context of the project. But this one is a bit more obscure. Specially in this case because it's a name and not an object per se

File: pkg/providers/vsphere/vsphere.go
Comment (g-gaston) [inline pkg/providers/vsphere/vsphere.go:937]: Just for my understanding, why don't we allow to reference a `VsphereMachineConfig` that already exists? I'm assuming that if it already exits it's because it was created by being referenced by a different eks-a cluster

A `VsphereMachineConfig` is just a "template` for a VM configuration, why couldn't it be shared by multiple clusters?

**Review by danbudris — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/567#pullrequestreview-797059556" title="Approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/567#" title="Author self-approved">jiayiwang7</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,jiayiwang7]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere.go
Comment (danbudris) [reply pkg/providers/vsphere/vsphere.go:931]: yeah I did find it kind of hard to read myself tbh

File: pkg/providers/vsphere/vsphere.go
Comment (jiayiwang7) [reply pkg/providers/vsphere/vsphere.go:937]: @mdsgabriel and i chatted offline. for the first path we dont want to allow sharing machineconfigs, especially btw management and workload cluster. e.g. if user upgrade workload cluster with shared machine config like network,  all machines in the management will have to be recreated. 
also the current delete workload cluster workflow deletes all eksa resources associated with it. If we allow sharing, it will also remove management's resources.

we can iterate after this release to be less restricted. but for now we add this validation to prevent unexpected behaviors

Comment (mdsgabriel) [reply pkg/providers/vsphere/vsphere.go:937]: > @mdsgabriel and i chatted offline. for the first path we dont want to allow sharing machineconfigs, especially btw management and workload cluster. e.g. if user upgrade workload cluster with shared machine config like network, all machines in the management will have to be recreated. also the current delete workload cluster workflow deletes all eksa resources associated with it. If we allow sharing, it will also remove management's resources.
> 
> we can iterate after this release to be less restricted. but for now we add this validation to prevent unexpected behaviors

yes. we can change the in the future

**Pull request merged**
